### PR TITLE
allow encoding and decoding of integers larger than 32 bits

### LINKED ISSUE #195: allow encoding and decoding of values larger than 32 bits
It's possible for the library to be asked to encode and decode values which occupy more than 32 bits.  An example of this is the WMO standard Table D BUFR sequence 3-15-010, where an operator is used to encode the constituent Table B BUFR descriptor 0-22-069 (spectral wave density) with 38 bits.

Currently, the only way to encode and decode such a value is by using the _8 build build of the library, which defaults to 8-bytes for **all** declared integers.  However, since a long-term goal is to consolidate all of the various _4, _8, and _d builds down to one (and, ideally, to the equivalent of the current _d build - see #78), then we need to figure out a way to encode values >32 bits in **any** of the current builds, by drilling down into the Fortran code to figure out what internal variables need to be explicitly declared as 8-byte integers, while continuing to allow everything else in the code to be declared as 4-byte integers.  This will allow for maximum library efficacy and portability in the future.

### PR DESCRIPTION
**Pull request by jbathegit**
Fixes #195

Part of #33

Part of #78

@jack-woollen please take a final look at this package and let me know if you have any questions.  As discussed in #195, I've just added a commit with a new f90 test code which tests the reading of integers larger than 32 bits.  Furthermore, this new test code also includes some additional tests for routines in the library which weren't already being tested.

At any rate, since I was the one who submitted this PR, I think you'll need to be the one to formally review and approve it before we can merge it ;-)

### CODE REVIEW

**Comment by jack-woollen** [comment]
@jbathegit  Since I was the one who authored the changes which "allow encoding and decoding of integers larger than 32 bits" (in all three versions of the bufrlib I might add),  then I should be the author of the pull request of the same name which will put these changes into the bufrlib. This becomes a record of who did what to contribute to the project. I appreciate your assistance in getting this ready to implement, but the bulk of the effort to actually make this work was mine, and that should be acknowledged by authorship of the PR. If you want individual credit for what you added to the jw64ints branch, beyond reviewing my changes, ie the correction and addition of various regression tests, then you should separate those changes into another branch and make a PR for that. I'll be happy to approve that PR for you, as I assume you will be to approve my PR for this significant upgrade of the bufrlib, which also, de facto, gives a clear path to enable the goal of reducing the lib down to a single build.

**Comment by jbathegit** [comment]
@jack-woollen if you'd like to submit a separate PR for all of these changes together, then I'm fine with that and can just delete this one.  In my mind, this was never about who gets credit for what, but just moving the process along.  And no matter who submits the PR, your name will still forever remain on all of the underlying commits that you contributed, as well as in the docblocks of the routines themselves, not to mention in the chat history for this issue within GitHub.

That said, I want you to get all of the credit that you rightly deserve, so if in your mind that includes being the one to submit the PR, then by all means please go ahead and do so and I'll approve it.  It's fine with me if all of the changes are together, including the two commits I added on top - those don't have to be a separate PR as far as I'm concerned.  Again, I'm not looking for personal recognition for myself here.  I totally agree that you did the bulk of the work on this issue, which (I believe?) I already acknowledged several times in the chat history of #195, so it definitely wasn't my intent here to steal your thunder in any way, but rather just to keep the process moving.  And that's why I went ahead and submitted a PR, because my last commit yesterday with the new regression test was (I believe?) the last piece we were both waiting on to put this issue to bed.

Bottom line: I sincerely appreciate all of your help, and sorry if that wasn't clear or if it seemed otherwise.  So please go ahead and submit your own PR, and when you do so please just include the "Fixes" and "Part of" lines shown above within your own text box, so that your PR automatically links to those issues, and in turn automatically closes #195 once it's merged.  Thanks!

**Comment by jack-woollen** [comment]
Closing this PR in order to to reopen it as the primary author.
